### DISCUSSION TITLE
Service Endpoints

### DISCUSSION THREAD

**Discussion by ShifanilaSenthil** [serenity-is/Serenity#7095] [Help]
Hi , can we deploy/ host the endpoint controller separetely as API in serenity framework?

**Comment by VictorTomaili** [comment]
Yes, you can separate the endpoints to different project. But you cannot use benefits of the source generation on the client side.

**Reply by rolembergfilho** [reply]
can we have examples to how to do it?

**Reply by ShifanilaSenthil** [reply]
Thank you for your reply!..Please guide me through this how can it be possible

**Reply by VictorTomaili** [reply]
No, we don't have any example.

**Comment by LSNicholls** [comment]
@ShifanilaSenthil this would be similar to calling the endpoints from *any* external application, wouldn't it? Including non-Serenity, non-Asp.Net, etc.  Having done this, I would say the only hard part was dealing with the credentials/permissions to the endpoint.  I can't give you an example of this because it would really depend a lot on the purpose of your service call.
